Library/Homebrew/global.rb

```diff
@@ -133,4 +135,4 @@ def auditing?
 require "tap"
 require "tap_constants"
 
-require "compat" if !ARGV.include?("--no-compat") && !ENV["HOMEBREW_NO_COMPAT"]
+require "compat" unless Homebrew::EnvConfig.no_compat?
```

Comment:
Do we want this as `EnvConfig` only or as `Homebrew.args` with a `env` switch?

Reply:
> Do we want this as `EnvConfig` only

This I think. I don't see a use of `--no-compat` outside of `brew tests` and it's setting `HOMEBREW_NO_COMPAT` there explicitly already.